public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,537 @@
+}
+
+export class SceneGridRow extends SceneObjectBase<SceneGridRowState> {
+  public static Component = SceneGridRowRenderer;
+
+  public constructor(state: SceneGridRowState) {
+    super({
+      isResizable: false,
+      isDraggable: true,
+      isCollapsible: true,
+      ...state,
+      size: {
+        ...state.size,
+        x: 0,
+        height: 1,
+        width: GRID_COLUMN_COUNT,
+      },
+    });
+  }
+
+  public onCollapseToggle = () => {
+    if (!this.state.isCollapsible) {
+      return;
+    }
+
+    const layout = this.parent;
+
+    if (!layout || !(layout instanceof SceneGridLayout)) {
+      throw new Error('SceneGridRow must be a child of SceneGridLayout');
+    }
+
+    layout.toggleRow(this);
+  };
+}
+
+function SceneGridRowRenderer({ model }: SceneComponentProps<SceneGridRow>) {
+  const styles = useStyles2(getSceneGridRowStyles);
+  const { isCollapsible, isCollapsed, isDraggable, title } = model.useState();
+  const layout = model.getLayout();
+  const dragHandle = <SceneDragHandle layoutKey={layout.state.key!} />;
+
+  return (
+    <div className={styles.row}>
+      <div className={cx(styles.rowHeader, isCollapsed && styles.rowHeaderCollapsed)}>
+        <div onClick={model.onCollapseToggle} className={styles.rowTitleWrapper}>
+          {isCollapsible && <Icon name={isCollapsed ? 'angle-right' : 'angle-down'} />}
+          <span className={styles.rowTitle}>{title}</span>
+        </div>
+        {isDraggable && isCollapsed && <div>{dragHandle}</div>}
+      </div>
+    </div>
+  );
+}
+
+const getSceneGridRowStyles = (theme: GrafanaTheme2) => {
+  return {
+    row: css({
+      width: '100%',
+      height: '100%',
+      position: 'relative',
+      zIndex: 0,
+      display: 'flex',
+      flexDirection: 'column',
+    }),
+    rowHeader: css({
+      width: '100%',
+      height: '30px',
+      display: 'flex',
+      justifyContent: 'space-between',
+      marginBottom: '8px',
+      border: `1px solid transparent`,
+    }),
+    rowTitleWrapper: css({
+      display: 'flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+    }),
+    rowHeaderCollapsed: css({
+      marginBottom: '0px',
+      background: theme.colors.background.primary,
+      border: `1px solid ${theme.colors.border.weak}`,
+      borderRadius: theme.shape.borderRadius(1),
+    }),
+    rowTitle: css({
+      fontSize: theme.typography.h6.fontSize,
+      fontWeight: theme.typography.h6.fontWeight,
+    }),
+  };
+};
+
+function validateChildrenSize(children: SceneLayoutChild[]) {
+  if (
+    children.find(
+      (c) =>
+        !c.state.size ||
+        c.state.size.height === undefined ||
+        c.state.size.width === undefined ||
+        c.state.size.x === undefined ||
+        c.state.size.y === undefined
+    )
+  ) {
+    throw new Error('All children must have a size specified');
+  }
+}
+
+// Source: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/dashboard/components/grid/utils.js#L28
```

Comment:
Do we need to merge the generated background being a reference to Metabase? I'd suggest removing it since we're mostly done with all functionalities. WDYT?

Reply:
yeah, we can get rid of it.